### PR TITLE
Get WooPay adapted extensions from server

### PR DESCRIPTION
**Pull request by alefesouza**
#### Changes proposed in this Pull Request

This PR updates the WooPay incompatible extensions, adapted extensions and available countries list from server with a single request.

This PR needs to be tested alongside 3568-gh-Automattic/woocommerce-payments-server.

#### Testing instructions

* Follow 3568-gh-Automattic/woocommerce-payments-server testing instructions.
* Follow [#6608](https://github.com/Automattic/woocommerce-payments/pull/6608) testing instructions changing `validate_incompatible_extensions` to `validate_woopay_compatibility` to ensure everything about incompatible extensions is working.
* Install the Points & Rewards plugin.
* Modify the lines [here](https://github.com/Automattic/woocommerce-payments-server/pull/3568/files#diff-5bb125880899ef7de4311905c5552402c73c1d7640a6c6e9d8bca2447ec86c0fR71) to look like this: `'adapted_extensions' => ['woocommerce-points-and-rewards'], 'available_countries' => ['US', 'BR'],` 
* Run `validate_woopay_compatibility` on the cron plugin.
* Access [phpMyAdmin](http://localhost:8083/index.php?route=/table/sql&db=wordpress&table=wp_options) and run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woocommerce_woocommerce_payments_woopay_available_countries'
```

* The `meta_value` should match the value you filled out.
* Run this query:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woopay_enabled_adapted_extensions'
```

* The `meta_value` should contain `woocommerce-points-and-rewards`.
* Uninstall or deactivate Points & Rewards plugin.
* Run `validate_woopay_compatibility` on the cron plugin.
* Run this query on phpMyAdmin:

```
SELECT * FROM `wp_options` WHERE `option_name` = 'woopay_enabled_adapted_extensions'
```

* The `meta_value` should not contains `woocommerce-points-and-rewards`.
* Deactivating or activating Points & Rewards plugin without running the cron should also update the `woopay_enabled_adapted_extensions` value.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6562` or branch name `add/add-single-woopay-compatibility-endpoint` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/add-single-woopay-compatibility-endpoint) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 66fc73b6533e9d81e417e988e63c142bcf294713
- Build time: 2023-07-18 18:31:18 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/woopay/class-woopay-scheduler.php [hunk 1]
```diff
@@ -52,51 +55,104 @@ public function init() {
 	 * Disables the scheduler when the plugin is disabled.
 	 */
 	public function remove_scheduler() {
-		wp_clear_scheduled_hook( 'validate_incompatible_extensions' );
+		wp_clear_scheduled_hook( 'validate_woopay_compatibility' );
```
Comment (malithsen) [inline includes/woopay/class-woopay-scheduler.php:59]: `validate_incompatible_extensions` schedule may still be running on sites that had the previous changes. Should we clear it out here or when upgrading the plugin?

File: includes/woopay/class-woopay-scheduler.php [hunk 2]
```diff
@@ -52,51 +55,104 @@ public function init() {
...
+	 * Update the enable adapted extensions list.
+	 *
+	 * @param array $adapted_extensions The adapted extensions list.
+	 * @param array $active_plugins     The active plugins.
+	 */
+	public function update_enabled_adapted_extensions( $adapted_extensions, $active_plugins ) {
+		try {
+			$enabled_adapted_extensions = [];
+
+			foreach ( $active_plugins as $plugin ) {
```
Comment (malithsen) [inline includes/woopay/class-woopay-scheduler.php]: This logic seems similar to the `contains_extensions_in_list` function. Can we call the same function here?

File: includes/woopay/class-woopay-scheduler.php [hunk 3]
```diff
@@ -106,30 +162,33 @@ public function show_warning_when_incompatible_extension_is_enabled( $plugin ) {
...
-		if ( ! $this->contains_incompatible_extension( $active_plugins, $incompatible_extensions ) ) {
+		if ( ! $this->contains_extensions_in_list( $active_plugins, $incompatible_extensions ) ) {
 			delete_option( self::INVALID_EXTENSIONS_FOUND_OPTION_NAME );
 		}
+
+		$this->update_enabled_adapted_extensions( $adapted_extensions, $active_plugins );
 	}
 
 	/**
 	 * Checks if there is any incompatible extension in the list.
```
Comment (malithsen) [inline includes/woopay/class-woopay-scheduler.php]: In a comment above I suggested reusing `contains_extensions_in_list` for adapted extensions. If we do end up doing that, let's update the doc string here to remove the 'incompatible extension' reference and make it more generic.

**Review by malithsen — COMMENTED** [review]
Changes looks good and tests well. Left some minor feedback in comments.

File: includes/woopay/class-woopay-scheduler.php (hunk 2 above)
Comment (alefesouza) [reply includes/woopay/class-woopay-scheduler.php]: Fixed on https://github.com/Automattic/woocommerce-payments/pull/6562/commits/9b1915db38b4b5e0686518ca8b691c9f19a611eb.

File: includes/woopay/class-woopay-scheduler.php (hunk 3 above)
Comment (alefesouza) [reply includes/woopay/class-woopay-scheduler.php]: Fixed on https://github.com/Automattic/woocommerce-payments/pull/6562/commits/9b1915db38b4b5e0686518ca8b691c9f19a611eb.

File: includes/woopay/class-woopay-scheduler.php (hunk 1 above)
Comment (alefesouza) [reply includes/woopay/class-woopay-scheduler.php:59]: Fixed on https://github.com/Automattic/woocommerce-payments/pull/6562/commits/24cb901f6d381c35b1451c0c51fdd489addbdbfa.

**Review by malithsen — APPROVED** [review]
Changs look good. Thanks @alefesouza

**Pull request merged**